### PR TITLE
rework workflow and add darwin index

### PR DESCRIPTION
**Pull request by Artturin**

### CODE REVIEW

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -1,22 +1,51 @@
...
+    - name: download darwin artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: darwin
+        path: darwin-index
+
     - name: Get current date
       id: date
-      run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+      run: echo "::set-output name=date::$(date +'%Y-%m-%d-%S')"
```
Comment (Mic92) [inline .github/workflows/update.yml:48]: Do we need seconds here?

```suggestion
      run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
```

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -27,13 +56,23 @@ jobs:
...
-      id: upload-release-asset
+    - name: Upload Linux Release Asset
+      id: upload-linux-asset
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ steps.create_release.outputs.upload_url }}
+        asset_path: linux-index/files
+        asset_name: linux-index
```
Comment (Mic92) [inline .github/workflows/update.yml:67]: If break compatibility maybe we should just include the architecture as well:

```suggestion
        asset_name: index-x86_64-linux
```

File: .github/workflows/update.yml [hunk 3]
```diff
@@ -27,13 +56,23 @@ jobs:
...
+      id: upload-darwin-asset
       uses: actions/upload-release-asset@v1
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
         upload_url: ${{ steps.create_release.outputs.upload_url }}
-        asset_path: ./files
-        asset_name: files
+        asset_path: darwin-index/files
+        asset_name: darwin-index
```
Comment (Mic92) [inline .github/workflows/update.yml:77]:
```suggestion
        asset_name: index-x86_64-darwin
```

File: .github/workflows/update.yml (hunk 1 above)
Comment (Artturin) [reply .github/workflows/update.yml]: had it for testing

**Pull request merged**